SU-HSDO/suhumsci
# SHS-5811: Add "logout" button to sites

# READY FOR REVIEW

## Summary
- Updates composer.json to point to a forked repo of the `stanford_samlauth` module that adds an optional logout button.

## Need Review By (Date)
2024-09-30

## Urgency
low

## Steps to Test
1. On a logged out page of the site, verify that the login button still exists in the footer.

![Screenshot 2024-09-26 at 5 02 59 PM](https://github.com/user-attachments/assets/880f2a85-8187-49b1-b7fd-cf53c2928fa4)

2. Login to the site
3. Visit the "SAML SUNetID Block" block configuration page either by going to the "Block layout page" and clicking "Configure" next to the "SAML SUNetID Block" block or hovering over the block on the front-end and clicking "Configure" in the context menu.
4. Verify there are 2 new fields: "Enable logout button" and "SUNETID log-out link text"

![Screenshot 2024-09-26 at 5 13 36 PM](https://github.com/user-attachments/assets/d6e83474-6f8b-4733-a798-11355e0e6fbb)

5. Test enabling and disabling the logout button and verify on the front-end that the logout button appears and disappears.

![Screenshot 2024-09-26 at 5 16 59 PM](https://github.com/user-attachments/assets/7d0be775-1bde-4779-ae8e-0ff329b8b945)

6. Test changing the logout button text and verify on the front-end that the button text changes.
7. Test using the logout button and verify that it correctly logs the user out of the system. If you were on a specific page, after logout, the system will attempt to redirect to the page you were on.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)
